entry.go

```diff
@@ -82,7 +82,10 @@ func (entry *Entry) WithFields(fields Fields) *Entry {
 	for k, v := range fields {
 		data[k] = v
 	}
-	return &Entry{Logger: entry.Logger, Data: data}
+	// shallow copy
+	newentry := *entry
+	newentry.Data = data
+	return &newentry
```

Comment:
I don't think the extra fields are going to cause a large performance issue compared with the cost of the allocation. What I am worried about is the errant copying causing a future race. The shared buffer also has cause for concern. It might be better/safer to move to a buffer pool.